@dschult Should I open an issue about the linprog failures in CI?

<details>

```
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPDense::test_sparse_constraints - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_docstring_example - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_sparse_constraints - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_inequality_constraints - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_inequality_constraints2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bounds_mixed - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bounds_equal_no_presolve - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_zero_column_1 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_zero_row_1 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_zero_row_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_singleton_row_eq_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_singleton_row_ub_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_infeasible - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_unbounded - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_cyclic_recovery - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_cyclic_bland - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_nontrivial_problem - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_lpgen_problem - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_network_flow - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_simplex_algorithm_wikipedia_example - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_enzo_example - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_enzo_example_b - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_enzo_example_c_with_degeneracy - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_enzo_example_c_with_infeasibility - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_basic_artificial_vars - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_optimize_result - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_5400 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_6139 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_7044 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_7237 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_8174 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_8174_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_8561 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_8662 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_10124 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparseCholmod::test_bug_10349 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_docstring_example - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_sparse_constraints - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_inequality_constraints - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_inequality_constraints2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bounds_mixed - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bounds_equal_no_presolve - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_zero_column_1 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_zero_row_1 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_zero_row_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_singleton_row_eq_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_singleton_row_ub_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_infeasible - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_unbounded - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_cyclic_recovery - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_cyclic_bland - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_nontrivial_problem - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_lpgen_problem - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_network_flow - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_simplex_algorithm_wikipedia_example - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_enzo_example - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_enzo_example_b - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_enzo_example_c_with_degeneracy - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_basic_artificial_vars - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_optimize_result - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_5400 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_7044 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_7237 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_8174 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_8174_2 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_8561 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_8662 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_10124 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_10349 - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
FAILED scipy/optimize/tests/test_linprog.py::TestLinprogIPSpecific::test_solver_select - scipy.optimize._optimize.OptimizeWarning: Solving system with option 'cholesky':True failed. It is normal for this to happen occasionally, especially as the solution is approached. However, if you see this frequently, consider setting option 'cholesky' to False.
= 73 failed, 58625 passed, 3747 skipped, 191 xfailed, 14 xpassed in 201.72s (0:03:21) =
```

</details>

--- a/scipy/optimize/_linprog_ip.py
+++ b/scipy/optimize/_linprog_ip.py
@@ -94,7 +94,7 @@ def _get_solver(M, sparse=False, lstsq=False, sym_pos=True,
                     _get_solver.cholmod_factor.cholesky_inplace(M)
                 except Exception:
                     with np.testing.suppress_warnings() as sup:
-                        sup.filter(CholmodTypeConversionWarning, "converting matrix of class")
+                        sup.filter(CholmodTypeConversionWarning, "converting *of class")
                         _get_solver.cholmod_factor = cholmod_analyze(M)
                         _get_solver.cholmod_factor.cholesky_inplace(M)
                 solve = _get_solver.cholmod_factor

--- a/scipy/optimize/_linprog_ip.py
+++ b/scipy/optimize/_linprog_ip.py
@@ -88,13 +88,13 @@ def _get_solver(M, sparse=False, lstsq=False, sym_pos=True,
                 def solve(r, sym_pos=False):
                     return sps.linalg.lsqr(M, r)[0]
             elif cholesky:
-                try:
-                    # Will raise an exception in the first call,
-                    # or when the matrix changes due to a new problem
-                    _get_solver.cholmod_factor.cholesky_inplace(M)
-                except Exception:
-                    with np.testing.suppress_warnings() as sup:
-                        sup.filter(CholmodTypeConversionWarning, "converting *of class")
+                with np.testing.suppress_warnings() as sup:
+                    sup.filter(CholmodTypeConversionWarning)
+                    try:
+                        # Will raise an exception in the first call,
+                        # or when the matrix changes due to a new problem
+                        _get_solver.cholmod_factor.cholesky_inplace(M)
+                    except Exception:
                         _get_solver.cholmod_factor = cholmod_analyze(M)
                         _get_solver.cholmod_factor.cholesky_inplace(M)
                 solve = _get_solver.cholmod_factor

--- a/scipy/optimize/_linprog_ip.py
+++ b/scipy/optimize/_linprog_ip.py
@@ -31,6 +31,7 @@ try:
     import sksparse  # noqa: F401
     from sksparse.cholmod import cholesky as cholmod  # noqa: F401
     from sksparse.cholmod import analyze as cholmod_analyze
+    from sksparse.cholmod import CholmodTypeConversionWarning
 except ImportError:
     has_cholmod = False
 try:
@@ -92,8 +93,10 @@ def _get_solver(M, sparse=False, lstsq=False, sym_pos=True,
                     # or when the matrix changes due to a new problem
                     _get_solver.cholmod_factor.cholesky_inplace(M)
                 except Exception:
-                    _get_solver.cholmod_factor = cholmod_analyze(M)
-                    _get_solver.cholmod_factor.cholesky_inplace(M)
+                    with np.testing.suppress_warnings() as sup:
+                        sup.filter(CholmodTypeConversionWarning, "converting matrix of class")
+                        _get_solver.cholmod_factor = cholmod_analyze(M)
+                        _get_solver.cholmod_factor.cholesky_inplace(M)
                 solve = _get_solver.cholmod_factor
             else:
                 if has_umfpack and sym_pos:

